procsim/biomass/level1_product_generator.py

```diff
@@ -322,75 +323,75 @@ def generate_output(self):
             self._hdrs.append(self._hdr)
 
         for hdr in self._hdrs:
-            self._generate_level1_stacked_product(hdr)
+            self._hdr = hdr
+            self._generate_level1_stacked_product()
 
-    def _generate_level1_stacked_product(self, hdr: main_product_header.MainProductHeader):
+    def _generate_level1_stacked_product(self):
 
         # Sanity check
-        if hdr.begin_position is None or hdr.end_position is None:
+        if self._hdr.begin_position is None or self._hdr.end_position is None:
```

Comment:
Now I see this, I'm not sure if it makes sense here. It was added for the case that no input products are available (see line 321), maybe the check must be moved to there.
(note that this is existing code, has nothing to do with the current changes)

Reply:
I have moved these checks near line 321 in 7f02b4e. Hopefully it makes more sense now, but feel free to move it someplace else still if you find a better spot.